langserver/langserver_test.go

```diff
@@ -912,6 +912,23 @@ type Header struct {
 			},
 		},
 	},
+	"go hover docs special cases": {
+		rootURI: "file:///src/test/pkg",
+		fs: map[string]string{
+			"q.go": `package p
+type T struct {
+	Q string // Q is a string field.
+	// X is documented.
+	X int // X has comments.
+}`,
+		},
+		cases: lspTestCases{
+			wantHover: map[string]string{
+				"q.go:3:2": "struct field Q string; Q is a string field. \n\n",
+				"q.go:5:2": "struct field X int; X is documented. \n\nX has comments. \n\n",
```

Comment:
should we be stripping whitespace on the right? Either way this looks good and is easy to change :)